cpp/tests/test_d_abm_model.cpp

```diff
@@ -0,0 +1,142 @@
+/* 
+* Copyright (C) 2020-2024 German Aerospace Center (DLR-SC)
+*
+* Authors: Julia Bicker
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "d_abm/parameters.h"
+#include "d_abm/quad_well.h"
+#include "d_abm/model.h"
+#include "d_abm/simulation.h"
+#include "memilio/utils/random_number_generator.h"
+#include "abm_helpers.h"
+#include <cstddef>
+#include <gtest/gtest.h>
+#include <vector>
+
+enum class InfectionState
+{
+    S,
+    E,
+    C,
+    I,
+    R,
+    D,
+    Count
+
+};
+
+TEST(TestQuadWell, well)
+{
+    auto well = well_index(Eigen::Vector2d{-1, 1});
+    EXPECT_EQ(well, size_t(0));
+}
+
+TEST(TestQuadWell, adopt)
+{
+    QuadWellModel<InfectionState>::Agent agent{Eigen::Vector2d{-1, 1}, InfectionState::S};
+    QuadWellModel<InfectionState> qw({agent}, {});
+    EXPECT_EQ(agent.status, InfectionState::S);
+    qw.adopt(agent, InfectionState::E);
+    EXPECT_EQ(agent.status, InfectionState::E);
+}
+
+TEST(TestQuadWell, adoptionRate)
+{
+    QuadWellModel<InfectionState>::Agent a1{Eigen::Vector2d{-1, 1}, InfectionState::S};
+    QuadWellModel<InfectionState>::Agent a2{Eigen::Vector2d{-1.2, 1}, InfectionState::I};
+    QuadWellModel<InfectionState> qw({a1, a2}, {{InfectionState::S,
+                                                 InfectionState::E,
+                                                 mio::dabm::Region(0),
+                                                 0.1,
+                                                 {InfectionState::C, InfectionState::I},
+                                                 {1, 0.5}}});
+    EXPECT_EQ(qw.adoption_rate(a1, InfectionState::E), 0.025);
```

Comment:
Other branches that are not checked are
- the rate for a2 to E is 0
- another isolated infected does not influence the rate
- the rate of a1 scales correctly with more nearby infected and non-infected

Reply:
Can you add descriptions to the test what it is you check? Otherwise it is hard to tell what is going on here.

This is true for all added tests. For example, in TestQuadWell.well(_index ?) one comment at the top like "check that well_index returns indices in the correct order" would be enough.
Here, in TestQuadWell.adoptionRate, you could write down what the agents and rates are used for, and/or explain what the EXPECTs do. Like "check that a1 is affected by a2, but not the for off a3" for the first EXPECT call, and so on.